docs/docs/path_algorithm.md

```diff
@@ -0,0 +1,406 @@
+
+* The relationships' property whose sum you want to minimize (`weight`).
+
+* A optional relationships' property whose sum you want to bound (`cost`) and the optional bound (`maxCost`).
+
+* An optional bound on the path length - the number of relationships along the path (`maxLen`).
+
+* The number of paths you want to retrieve: either all minimal-weight paths (`pathCount` is 0), a single minimal-weight path (`pathCount` is 1), or _n_ minimal-weight paths with potentially different weights (`pathCount` is _n_).
+
+This topic explains which problems you can solve using these algorithms and demonstrates how to use them.
+
+Let's start with the following graph.
+
+![Road network](../images/road_network.png)
+
+This graph represents a road network with 7 cities (A, B, C, and so on) and 11 one-way roads. Each road has a distance (say, in kilometers) and trip time (say, in minutes).
+
+Let's create the graph.
+
+{{< highlight bash >}}
+GRAPH.QUERY g "CREATE (a:City{name:'A'}), (b:City{name:'B'}), (c:City{name:'C'}), (d:City{name:'D'}), (e:City{name:'E'}), (f:City{name:'F'}), (g:City{name:'G'}), (a)-[:Road{time:4, dist:3}]->(b), (a)-[:Road{time:3, dist:8}]->(c), (a)-[:Road{time:4, dist:2}]->(d), (b)-[:Road{time:5, dist:7}]->(e), (b)-[:Road{time:5, dist:5}]->(d), (d)-[:Road{time:4, dist:5}]->(e), (c)-[:Road{time:3, dist:6}]->(f), (d)-[:Road{time:1, dist:4}]->(c), (d)-[:Road{time:2, dist:12}]->(f), (e)-[:Road{time:5, dist:5}]->(g), (f)-[:Road{time:4, dist:2}]->(g)"
+ {{< / highlight >}}
+
+If you're using RedisInsight v2, you can create and visualize the graph by slightly modifying the above query: you'll have to assign aliases to all nodes and relationships, and return them:
+
+{{< highlight bash >}}
+GRAPH.QUERY g "CREATE (a:City{name:'A'}), (b:City{name:'B'}), (c:City{name:'C'}), (d:City{name:'D'}), (e:City{name:'E'}), (f:City{name:'F'}), (g:City{name:'G'}), (a)-[r1:Road{time:4, dist:3}]->(b), (a)-[r2:Road{time:3, dist:8}]->(c), (a)-[r3:Road{time:4, dist:2}]->(d), (b)-[r4:Road{time:5, dist:7}]->(e), (b)-[r5:Road{time:5, dist:5}]->(d), (d)-[r6:Road{time:4, dist:5}]->(e), (c)-[r7:Road{time:3, dist:6}]->(f), (d)-[r8:Road{time:1, dist:4}]->(c), (d)-[r9:Road{time:2, dist:12}]->(f), (e)-[r10:Road{time:5, dist:5}]->(g), (f)-[r11:Road{time:4, dist:2}]->(g) return a,b,c,d,e,f,g,r1,r2,r3,r4,r5,r6,r7,r8,r9,r10,r11"
+{{< / highlight >}}
+
+![Road network](../images/graph_query_city.png)
+
+## Before RedisGraph 2.10
+
+Before 2.10, you were able to solve these queries:
+
+* Find the shortest path (by number of roads) from A to G
+* Find all the shortest paths (by number of roads) from A to G
+* Find 5 shortest paths (by number of roads) from A to G
+* Find 5 shortest paths (in kilometers) from A to G
+
+### Find the shortest path (by number of roads) from A to G
+
+{{< highlight bash >}}
+GRAPH.QUERY g "MATCH (a:City{name:'A'}),(g:City{name:'G'}) WITH shortestPath((a)-[*]->(g)) as p return length(p), [n in nodes(p) | n.name] as pathNodes"
```

Comment:
clause names should be all uppercased, `return` -> `RETURN`

Reply:
Indeed